docs/developer-env.md

```diff
@@ -0,0 +1,534 @@
+- [Nextflow](https://marketplace.visualstudio.com/items?itemName=nextflow.nextflow) - Nextflow language support.
+- [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) - Code formatter using Prettier.
+- [Rainbow CSV](https://marketplace.visualstudio.com/items?itemName=mechatroner.rainbow-csv) - Highlight columns in .CSV files in different colors.
+- [Ruff](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff) - An extremely fast Python linter and code formatter, written in Rust.
+- [Todo Tree](https://marketplace.visualstudio.com/items?itemName=Gruntfuggly.todo-tree) - Show TODO, FIXME, etc. comment tags in a tree view.
+- [YAML](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) - YAML Language Support by Red Hat, with built-in Kubernetes syntax support.
+- [Markdown Extended](https://marketplace.visualstudio.com/items?itemName=jebbs.markdown-extended) - Gives Markdown previews, including admonitions.
+
+````{tabs}
+
+```{group-tab} Windows
+
+To install the nf-core extension pack on Windows:
+
+1. Open VS Code.
+1. Open the **Extensions** view in the left-hand menu.
+1. Search for **nf-core-extensionpack**.
+1. Select **Install**.
+
+```
+
+```{group-tab} macOS
+
+To install the nf-core extension pack on macOS:
+
+1. Open VS Code.
+1. Open the **Extensions** view in the left-hand menu.
+1. Search for **nf-core-extensionpack**.
+1. Select **Install**.
+
+```
+
+```{group-tab} Linux
+
+To install the nf-core extension pack on Linux Debian/Ubuntu distributions:
+
+1. Open VS Code.
+1. Open the **Extensions** view in the left-hand menu.
+1. Search for **nf-core-extensionpack**.
+1. Select **Install**.
+
+```
+
+````
+
+(remote-development-ext)=
+
+### Remote Development
+
+The [Remote Development extension pack](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack) enables you to run WSL, SSH, or a development container for editing and debugging with the full set of VS Code features.
+
+The Remote Development extension pack contains 4 extensions:
+
+- [Remote - SSH](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh) - Work with source code in any location by opening folders on a remote machine/VM using SSH.
+- [Remote - Tunnels](https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-server) - Work with source code in any location by opening folders on a remote machine/VM using a VS Code Tunnel (rather than SSH).
+- [Dev Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) - Work with a separate toolchain or container based application by opening any folder mounted into or inside a container.
+- [WSL](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) - Get a Linux-powered development experience from the comfort of Windows by opening any folder in the Windows Subsystem for Linux.
```

Comment:
I'm a bit confused by these as they're not really remote..? Maybe worth addressing this given the heading?

Reply:
I think they are treated as remote file systems. Using the same language as used in the pack description. Main purpose is for windows developers and those accessing remote file systems.

Reply:
The VSCode experience in WSL is basically the same as native Windows

Also, this text seems overly promotional

Reply:
I agree the text is very promotional. I took the text from [here](https://code.visualstudio.com/docs/remote/remote-overview), I'll tone it down.